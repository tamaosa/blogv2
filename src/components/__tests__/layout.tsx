import React from "react"
import { useStaticQuery } from "gatsby"
import { render, fireEvent } from "@testing-library/react"
import Layout from "../layout"

describe("Layout", () => {
  const siteTitle = "siteTitle"
  const name = "tamaosa"
  const summary = "summary"
  const twitter = "twitter"
  const github = "github"
  beforeAll(() => {
    ;(useStaticQuery as jest.Mock).mockReturnValue({
      site: {
        siteMetadata: {
          title: siteTitle,
          author: {
            name: name,
            summary: summary,
          },
          social: {
            twitter: twitter,
            github: github,
          },
        },
      },
    })
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  it("snapshot test", () => {
    const { asFragment } = render(<Layout />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("testing link to home", () => {
    const { container } = render(<Layout />)
    expect(container.innerHTML).toMatch(`<a href="/">${siteTitle}</a>`)
  })

  it("testing onclick light/dark button", () => {
    const { container, getByText } = render(<Layout />)
    expect(container).toHaveTextContent("ðŸŒž")
    fireEvent.click(getByText("ðŸŒž"))
    expect(container).toHaveTextContent("ðŸŒ›")
    fireEvent.click(getByText("ðŸŒ›"))
    expect(container).toHaveTextContent("ðŸŒž")
  })

  it("testing link to privacy policy", () => {
    const { container } = render(<Layout />)
    expect(container.innerHTML).toMatch(
      `<a href="/privacy-policy">Privacy Policy</a>`
    )
  })

  it("testing copyright", () => {
    const { container } = render(<Layout />)
    expect(container).toHaveTextContent("Â© 2019 tamaosa")
  })
})
