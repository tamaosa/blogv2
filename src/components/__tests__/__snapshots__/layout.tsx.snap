// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot test 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    -webkit-transform: translate(0px, 0px);
    -moz-transform: translate(0px, 0px);
    -ms-transform: translate(0px, 0px);
    transform: translate(0px, 0px);
  }

  75% {
    -webkit-transform: translate(0px, 0px);
    -moz-transform: translate(0px, 0px);
    -ms-transform: translate(0px, 0px);
    transform: translate(0px, 0px);
  }

  80% {
    -webkit-transform: translate(0px, 2px);
    -moz-transform: translate(0px, 2px);
    -ms-transform: translate(0px, 2px);
    transform: translate(0px, 2px);
  }

  85% {
    -webkit-transform: translate(2px, 0px);
    -moz-transform: translate(2px, 0px);
    -ms-transform: translate(2px, 0px);
    transform: translate(2px, 0px);
  }

  90% {
    -webkit-transform: translate(0px, 2px);
    -moz-transform: translate(0px, 2px);
    -ms-transform: translate(0px, 2px);
    transform: translate(0px, 2px);
  }

  95% {
    -webkit-transform: translate(2px, 0px);
    -moz-transform: translate(2px, 0px);
    -ms-transform: translate(2px, 0px);
    transform: translate(2px, 0px);
  }

  100% {
    -webkit-transform: translate(0px, 0px);
    -moz-transform: translate(0px, 0px);
    -ms-transform: translate(0px, 0px);
    transform: translate(0px, 0px);
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  margin: 0 1.125rem;
}

@media (min-width: 46.5rem) {
  .emotion-0 {
    margin: 0 auto;
    width: 46.5rem;
  }
}

.emotion-0>header {
  margin-top: 0.375rem;
}

.emotion-0>main {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-top: 2.25rem;
  margin-bottom: 0.375rem;
}

.emotion-0>footer {
  margin-top: 0.375rem;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 a {
  color: var(--fg);
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Century Gothic,CenturyGothic,AppleGothic,sans-serif;
  font-weight: bold;
  font-size: 1.18921rem;
  line-height: 1.5rem;
}

.emotion-3 button {
  background: var(--bg);
  font-size: 1.41421rem;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-3 button:hover {
  -webkit-animation: animation-0 2s infinite;
  animation: animation-0 2s infinite;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.emotion-6 {
  margin: 0 0.75rem 0 0;
}

.emotion-7 {
  margin: 0;
}

.emotion-7 span {
  font-weight: bold;
}

.emotion-7 a {
  color: var(--fg);
}

.emotion-7 div {
  margin: 0;
  font-size: 0.917rem;
  line-height: 1.5rem;
}

.emotion-8 {
  display: inline-block;
  width: 1.125rem;
  margin-left: 0.375rem;
}

.emotion-10 {
  font-size: 0.917rem;
  margin-top: 0.75rem;
  text-align: right;
}

.emotion-10 small {
  display: block;
  margin: 0;
}

<div
    class="emotion-0"
  >
    <header
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          href="/"
        >
          siteTitle
        </a>
      </div>
      <div
        class="emotion-3"
      >
        <button>
          🌞
        </button>
      </div>
    </header>
    <main />
    <footer>
      <div
        class="emotion-4"
      >
        <div
          class="emotion-5"
        >
          <div
            class="emotion-6"
          >
            <img
              alt="プロフィール画像"
              height="72"
              layout="fixed"
              quality="95"
              src="../../content/images/avator.png"
              style="margin: 0px; padding: 0px; min-width: 50px; border-radius: 50%;"
              width="72"
            />
          </div>
          <div
            class="emotion-7"
          >
            <span>
              tamaosa
            </span>
            <a
              aria-label="GitHubアカウント"
              href="https://github.com/github"
              rel="external noopener"
              target="_blank"
            >
              <span
                class="emotion-8"
              >
                <i
                  class="fa"
                />
              </span>
            </a>
            <a
              aria-label="Twitterアカウント"
              href="https://twitter.com/twitter"
              rel="external noopener"
              target="_blank"
            >
              <span
                class="emotion-8"
              >
                <i
                  class="fa"
                />
              </span>
            </a>
            <div>
              summary
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-10"
      >
        <small>
          <a
            href="/privacy-policy"
          >
            Privacy Policy
          </a>
        </small>
        <small>
          Built with 
          <a
            href="https://www.gatsbyjs.com"
            rel="external noopener"
            target="_blank"
          >
            Gatsby
          </a>
          , © 2019 
          <a
            href="https://www.tamaosa.com"
          >
            tamaosa
          </a>
        </small>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
